utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,223 @@
+            address.city = cityElement.get(0).data();
+        }
+
+        if stateElement is xml:Element && stateElement.length() > 0 {
+            address.state = stateElement.get(0).data();
+        }
+
+        if postalCodeElement is xml:Element && postalCodeElement.length() > 0 {
+            address.postalCode = postalCodeElement.get(0).data();
+        }
+        return address;
+    }
+    return;
+}
+
+# Map C-CDA telecom to FHIR ContactPoint.
+#
+# + telecomElement - C-CDA telecom element
+# + return - Return FHIR ContactPoint
+public isolated function mapCcdaTelecomToFhirTelecom(xml<xml:Element> telecomElement) returns r4:ContactPoint? {
+    if telecomElement is xml:Element && telecomElement.length() > 0 {
+        r4:ContactPoint telecom = {};
+        string|error telecomUse = telecomElement.use;
+        string|error telecomValue = telecomElement.value;
+
+        if telecomValue is string {
+            telecom.value = telecomValue;
+        } else {
+            log:printDebug("Telecom value is not available", telecomValue);
+        }
+
+        if telecomUse is string {
+            match telecomUse {
+                "HP" => {
+                    telecom.use = "home";
+                }
+                "WP" => {
+                    telecom.use = "work";
+                }
+            }
+        } else {
+            log:printDebug("Telecom use is not available", telecomUse);
+        }
+        return telecom;
+    }
+    return;
+}
+
+# Map C-CDA name to FHIR HumanName.
+#
+# + nameElement - C-CDA name element
+# + isNamespaceAvailable - Check whether the namespace is available
+# + return - Return FHIR HumanName
+public isolated function mapCcdaNametoFhirName(xml<xml:Element> nameElement, boolean isNamespaceAvailable = true) returns r4:HumanName? {
+    if nameElement is xml:Element && nameElement.length() > 0 {
+        r4:HumanName name = {};
+        xml<xml:Element> familyElement;
+        xml<xml:Element> givenElement;
+
+        if isNamespaceAvailable {
+            familyElement = nameElement.children().<v3:family>;
+            givenElement = nameElement.children().<v3:given>;
+        } else {
+            familyElement = nameElement.children().<family>;
+            givenElement = nameElement.children().<given>;
+        }
+        if familyElement is xml:Element && familyElement.length() > 0 {
+            name.family = familyElement.get(0).data();
+        }
+
+        if givenElement is xml:Element && givenElement.length() > 0 {
+            name.given = [givenElement.get(0).data()];
+        }
+        return name;
+    }
+    return;
+}
+
+# Map C-CDA code to FHIR CodeableConcept.
+#
+# + codingElement - C-CDA code element
+# + return - Return FHIR CodeableConcept
+public isolated function mapCcdaCodingtoFhirCodeableConcept(xml<xml:Element> codingElement) returns r4:CodeableConcept? {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        r4:CodeableConcept codeableConcept = {};
```

Comment:
Do you need this to be defined here? Why not just create and return in-line?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/a6f2501164b72a5fead9204c9685a13db0160941